### PR TITLE
Add RELEASE.md

### LINKED ISSUE #888: CI: Improve and document releasing a new version
# The current workflow

1. Manually create a branch `releases/X.Y.Z` from `master`
    * Automatically updates the code to include the X.Y.Z version number
    * Automatically creates a vX.Y.Z tag
    * Automatically creates a PR to the charts repo
2. Manually merge PR on charts repo
3. Manually create a GitHub release from the existing `vX.Y.Z` tag

# A proposed workflow

- Manually create a branch `releases/X.Y.Z` either from `master` or an existing `releases/.X.Y.(Z-1)` branch
    - Automatically updates the code to include the X.Y.Z version number
    - Runs the Build workflow
-  When ready to publish the release create a GitHub release with `vX.Y.Z` tag from `releases/X.Y.Z`
   - Automatically creates a PR to the charts repo (automatically merge?)

In the future the last step should change to create `vX.Y.Z` tag from `releases/X.Y.Z` and from there goreleaser will be used (to also build and release the command line tool). 

# Tasks
- [x] Update CI workflows
- [ ] Document the release process

### PR DESCRIPTION
**Pull request by roee88**
Closes #888

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Release Process
+
+The process of creating a release is described in this document. Replace `X.Y.Z` with the version to be released.
+
+## 1. Create a `releases/X.Y.Z` branch 
+
+The `releases/X.Y.Z` branch should be created from a base branch. For minor releases the base is `master` and for patch releases the base is `releases/X.Y.(Z-1)`.
+
```
Comment (ronenkat) [inline RELEASE.md:8]: can we add a for example how to do it? `i.e., for this step use the github UI for switching branches to create a new branch.`

Comment (roee88) [reply RELEASE.md:8]: Added a link to GitHub docs on that

File: RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# Release Process
+
+The process of creating a release is described in this document. Replace `X.Y.Z` with the version to be released.
+
+## 1. Create a `releases/X.Y.Z` branch 
+
+The `releases/X.Y.Z` branch should be created from a base branch. 
+
+For minor releases the base is `master` and for patch releases the base is `releases/X.Y.(Z-1)`.
```
Comment (ronenkat) [inline RELEASE.md:9]:
```suggestion
For major and minor releases the base is `master` and for patch releases (fixes) the base is `releases/X.Y.(Z-1)`.
A new patch release should be created before merging pull-requests as described in the next section.
```

File: RELEASE.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+You can do that [directly from GitHub](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-and-deleting-branches-within-your-repository#creating-a-branch).
+
+## 2. Optionally create Pull Requests to `releases/X.Y.Z`
+
+For any change that is required on top of the base you should create a Pull Request targeting the release branch.
```
Comment (ronenkat) [inline RELEASE.md:15]:
```suggestion
For any change that is required on top of the base you should create a Pull Request targeting the **new** release branch. 
```

**Review by ronenkat — APPROVED** [review]
Added minor updates.

**Pull request merged**